### PR TITLE
M0269MD-21 Mock Framework

### PR DESCRIPTION
**Pull request by apineau-microej**
Add mock framework section: https://microej-docs--980.com.readthedocs.build/en/980/VEEPortingGuide/mock.html#mock-framework

### CODE REVIEW

File: VEEPortingGuide/mock.rst [hunk 1]
```diff
@@ -356,6 +356,225 @@ The Module serves two purposes, depending on whether it is added to a Mock or a
...
+or
+
+.. code-block:: java
+
+    @Property(getter = "com.microej.example.RandomService.getEnable", setter = "com.microej.example.RandomService.setEnable")
+    public class MyProperty extends BooleanProperty {
+        ...
+    }
+
+These attributes are optional. When not specified, the corresponding code will not be generated.
```
Comment (emoreac-microej) [inline VEEPortingGuide/mock.rst:469]:
```suggestion
These attributes are optional. When no attribute is specified, the corresponding code will not be generated. 
```

File: VEEPortingGuide/mock.rst [hunk 2]
```diff
@@ -356,6 +356,225 @@ The Module serves two purposes, depending on whether it is added to a Mock or a
...
+   MockupApplication.getProperty(MyProperty.class).setValue(!state);
+
+
+Mock Framework Widgets
+~~~~~~~~~~~~~~~~~~~~~~
+
+The Mock Framework provides some widgets to manipulate properties.
+
+Interacting with the widget modifies the underlying property,
+and in the contrary, updating the property value modifies the state of the widget.
```
Comment (emoreac-microej) [inline VEEPortingGuide/mock.rst:494]:
```suggestion
and similarly, updating the property value modifies the state of the widget.
```

File: VEEPortingGuide/mock.rst [hunk 3]
```diff
@@ -356,6 +356,225 @@ The Module serves two purposes, depending on whether it is added to a Mock or a
...
+         .. code-block:: kotlin
+
+            compileOnly(group="com.microej.tool", name="javafx", version="1.2.0", configuration="provided")
+
+      - Add the Mock Framework and JavaFX annotation processors dependencies to your Mock project:
+
+         .. code-block:: kotlin
+
+            annotationProcessor("com.microej.library.mock:mock-framework:1.0.1")
+            annotationProcessor(group="com.microej.tool", name="javafx", version="1.2.0", configuration="provided")
```
[the remark this answers is not in the document]
Comment (acolleux-microej) [reply VEEPortingGuide/mock.rst]: Fixed

File: VEEPortingGuide/mock.rst [hunk 4]
```diff
@@ -356,6 +356,225 @@ The Module serves two purposes, depending on whether it is added to a Mock or a
...
+   Mock Framework used to mock Heart Rate sensor on Wearable Demo.
+
+Usage
+-----
+
+The following steps should be followed to create a mock using Mock Framework:
+
+- Create a Mock Framework Property to bind the native method to mock to the mock UI,
+- Create a Dashboard, 
+- Add widgets.
```
[the remark this answers is not in the document]
Comment (acolleux-microej) [reply VEEPortingGuide/mock.rst]: I updated the paragraph, let me know if it's enough

**Review by emoreac-microej — APPROVED** [review]
I added a suggestion but it's now ok for me if you apply it or not, MR approved.

**Pull request merged**